### DISCUSSION TITLE
Ability to assign multiple authors for one entry

### DISCUSSION THREAD

**Discussion by shbchk** [ExpressionEngine/ExpressionEngine#2962] [Feature Requests]
Hi there! I would like to make a feature request.

I believe that one entry can have more than one author, so these authors can edit entry as their own. 

Maybe there can be one "primary author" and ability to select "additional authors", who will be able to edit\delete the entry as their own?

**Comment by litzinger** [comment]
Just my 2 cents: I don't think this should be core/native functionality. The need for multiple authors seems like a specific and rare use case. It would require a database update and a lot of add-ons expect there to be a single author for an entry. Not to mention all the internal querying of entries would need to be updated to query for 1 or more authors.

This should be a 3rd party fieldtype to select 1 or more authors, or if a native custom field that is simply a Relationship field where you can choose which members to display in the options list, then in templates you treat it as a normal Relationship field.

**Comment by robsonsobral** [comment]
> The need for multiple authors seems like a specific and rare use case.

I used to think like @litzinger until I discover how many hacks users do for it. Actually, it's far too common for newspapers to have entries co-written by two or three journalists.

However, he's right that it would be hard to implement. My suggestion is to allow us to have [members as options for fields](https://github.com/ExpressionEngine/ExpressionEngine/issues/1080).

**Comment by shbchk** [comment]
I agree, this use case is really not that rare. Co-authors and contributors are common for larger texts. 

And, often it is very helpful to have an opportunity to give a group of users permission to edit just one entry.

**Comment by robsonsobral** [comment]
@shbchk , in my experience, the "real" author is only one, most of the time it's an editor.

It would be really hard to implement it. Can you, please, tell us about your use case? It looks like it's more a matter of permissions than an informative field.

**Comment by shbchk** [comment]
In my cases it is a matter of permissions, yes. But as employee of a large newspaper, I can understand the needs of publishers.

I am currently developing 2 websites, and both of them can benefit from this feature.

First one is a [film festival database](https://vurchel.com), my personal project at very early stage :) In this case I would like to provide festival representatives and film authors opportunity to claim and manage related pages (entries).

Second one is a website for our volunteer organization, in which I would like to provide certain members of our organization ability to edit certain pages (entries). Currently I am thinking of using Notion <-> Wordpress integration to synchronize contents of Notion, where I can set any permissions for the pages, with Wordpress pages.

**Comment by robsonsobral** [comment]
Given that channels can share fields, can't you achieve what you need using a mix of channels and roles?

- channel `films_A` can be edited only by members of role **A**;
- channel `films_b` can be edited only by members of role **B**;

**Comment by shbchk** [comment]
Vurchel.com has thousands of entries. So, I will have to create thousands of channels and roles? I don't think this is good idea :)

**Comment by robsonsobral** [comment]
How many permissions? 4? 5? 60? To infinity and beyond? How many users?

It looks like a matter of an improvement on permissions, not authorship.

**Comment by litzinger** [comment]
If it's an issue with presenting more than 1 author, then that can be pretty easily solved with a custom fieldtype or enabling members to be selected as options in a checkbox group as @robsonsobral linked to.

If it's an issue with assigning more than 1 member to have editing permissions on existing entries then yes that gets more complicated and updates in EE would be rather far reaching I think. Things like the entry manager where you can filter by author gets more complex. Simply throwing an error or redirecting if a member tries to edit an entry that they are not the primary author of, or a related author needs to be taken into account. EE tables already have a single member_id relationship, which could remain as the primary author, but then it may need a new join table to assign additional sub-authors to it. How would this affect the channel:entries `author_id=""` parameter? How would it affect any channel:form code? I could see this being a rather large change in EE, and it would need to be done in a  way that doesn't break any add-on out there that refers to the `member_id` database column or tag parameters.

**Comment by shbchk** [comment]
Yes, it is the second one, permission issue. 

I can see that this is really complicated and requires a lot of changes, so probably I won't see it done :)

BTW, there are two addons providing ability to relate members to entries: [User](https://eeharbor.com/user/documentation/related_authors) and [GM Member Select](https://expressionengine.com/add-ons/gb-member-select).

**Comment by intoeetive** [comment]
We're planning to add Member fieldtype, allowing to select member from entry (I realize this is not same as authors though): https://github.com/ExpressionEngine/ExpressionEngine/pull/3031
